DuckDuckGo/Freemium/PIR/FreemiumPIRScanResultPolling.swift

```diff
@@ -0,0 +1,192 @@
+    ///   - dataManager: The data manager responsible for managing broker protection data.
+    ///   - freemiumPIRUserStateManager: Manages the state of the user's profile in Freemium PIR.
+    ///   - notificationCenter: The notification center used for posting and observing notifications. Defaults to `.default`.
+    ///   - timerInterval: The interval in seconds between polling checks. Defaults to 1 hour.
+    ///   - maxCheckDuration: The maximum time allowed before stopping polling without results. Defaults to 24 hours.
+    ///   - dateFormatter: A `DateFormatter` for formatting dates. Defaults to a POSIX date-time formatter.
+    init(
+        dataManager: DataBrokerProtectionDataManaging,
+        freemiumPIRUserStateManager: FreemiumPIRUserStateManager,
+        notificationCenter: NotificationCenter = .default,
+        timerInterval: TimeInterval = 3600,  // 1 hour in seconds
+        maxCheckDuration: TimeInterval = 86400,  // 24 hours in seconds
+        dateFormatter: DateFormatter = DefaultFreemiumPIRScanResultPolling.makePOSIXDateTimeFormatter()
+    ) {
+        self.dataManager = dataManager
+        self.freemiumPIRUserStateManager = freemiumPIRUserStateManager
+        self.notificationCenter = notificationCenter
+        self.timerInterval = timerInterval
+        self.maxCheckDuration = maxCheckDuration
+        self.dateFormatter = dateFormatter
+    }
+
+    /// Starts polling for PIR scan results or observes for a profile saved notification if no profile has been saved yet.
+    func startPollingOrObserving() {
+        if firstProfileSaved {
+            startRepeatingConditionCheck()
+        } else {
+            observeNotification()
+        }
+    }
+
+    deinit {
+        if let observer = observer {
+            notificationCenter.removeObserver(observer)
+        }
+        timer?.invalidate()
+    }
+}
+
+private extension DefaultFreemiumPIRScanResultPolling {
+
+    /// A Boolean value indicating whether the first profile has been saved.
+    var firstProfileSaved: Bool {
+        freemiumPIRUserStateManager.firstProfileSavedTimestamp != nil
+    }
+
+    /// The saved timestamp of the first profile as a `Date`, or `nil` if no profile has been saved yet.
+    var firstProfileSavedTimestamp: Date? {
+        get {
+            guard let timestampString = freemiumPIRUserStateManager.firstProfileSavedTimestamp else { return nil }
+            return dateFormatter.date(from: timestampString)
+        }
+        set {
+            if let newTimestamp = newValue {
+                let timestampString = dateFormatter.string(from: newTimestamp)
+                freemiumPIRUserStateManager.firstProfileSavedTimestamp = timestampString
+            } else {
+                freemiumPIRUserStateManager.firstProfileSavedTimestamp = nil
+            }
+        }
+    }
+
+    /// Observes the notification for when the first profile is saved and triggers the polling process.
+    func observeNotification() {
```

Comment:
Since it's only observing one specific notification I think it would be better to include it in the name (e.g. observeProfileSavedNotification)